nixos/modules/tasks/network-namespaces.nix

```diff
@@ -0,0 +1,44 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.networking;
+in
+{
+  options = {
+    networking.namespaces = mkOption {
+      default = [];
+      type = types.listOf types.str;
+      description = ''
+        A list of named network namespaces to create.
+      '';
+    };
+  };
+  config = mkIf (builtins.length cfg.namespaces != 0) {
```

Comment:
I think you should be doing one unit per network namespace. Otherwise failures in one only show up as failures for all of them. Being able to stop/start them individually might also be nice.

Reply:
done